### PR TITLE
Add missing builtin extensions

### PR DESCRIPTION
**Pull request by vince-fugnitto**
This commit adds the following:
- node-debug
- node-debug2
- references-view

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by marcdumais-work** [comment]
Note: "missing" in the sense these are used in the main repo's example applications and are not available for download anywhere "nice".

File: extensions.json [hunk 1]
```diff
@@ -283,6 +283,21 @@
...
       "repository": "https://github.com/Binaryify/OneDark-Pro"
+    },
+    {
+      "id": "ms-vscode.node-debug",
+      "version": "1.35.3",
+      "repository": "https://github.com/microsoft/vscode-node-debug"
+    },
+    {
+      "id": "ms-vscode.node-debug2",
+      "version": "1.33.0",
```
Comment (jankeromnes) [inline extensions.json]: Latest available version seems to be `1.42.1`: https://github.com/microsoft/vscode-node-debug2/releases

```suggestion
      "version": "1.42.1",
```

File: extensions.json [hunk 2]
```diff
@@ -283,6 +283,21 @@
       "id": "zhuangtongfa.material-theme",
       "version": "3.2.4",
       "repository": "https://github.com/Binaryify/OneDark-Pro"
+    },
+    {
+      "id": "ms-vscode.node-debug",
+      "version": "1.35.3",
```
Comment (jankeromnes) [inline extensions.json]: Why not the latest available version? https://github.com/microsoft/vscode-node-debug/releases

```suggestion
      "version": "1.41.1",
```

File: extensions.json [hunk 3]
```diff
@@ -283,6 +283,21 @@
...
+      "repository": "https://github.com/microsoft/vscode-node-debug"
+    },
+    {
+      "id": "ms-vscode.node-debug2",
+      "version": "1.33.0",
+      "repository": "https://github.com/microsoft/vscode-node-debug2"
+    },
+    {
+      "id": "ms-vscode.references-view",
+      "version": "0.0.47",
```
Comment (jankeromnes) [inline extensions.json]: Latest available version seems to be `0.0.50`: https://github.com/microsoft/vscode-references-view/blob/master/package.json#L6

```suggestion
      "version": "0.0.50",
```

**Comment by akosyakov** [comment]
@jankeromnes Latest version don't work in Theia or were not tested. We need these versions. Is it possible to publish 2 versions? 2 PRs?

**Comment by jankeromnes** [comment]
> @jankeromnes Latest version don't work in Theia or were not tested.

Thanks! Then it may be interesting to double-check the other published versions (we've generally picked only the latest available version until now).

> We need these versions.

Then please also specify the relevant `"checkout"` location, e.g. like so:

```json
    {
      "id": "ms-vscode.node-debug2",
      "version": "1.33.0",
      "repository": "https://github.com/microsoft/vscode-node-debug",
      "checkout": "v1.33.0"
    },
```

This will make the publishing script automatically checkout the `v1.33.0` before publishing the extension (otherwise, when `"checkout"` is not specified, it publishes straight from the latest `git clone`, regardless of what `"version"` says).

> Is it possible to publish 2 versions?

I'd say yes, because uniqueness is not enforced in `extensions.json`.

However, currently the script will compare `"version"` with the latest available on https://open-vsx.org, and it will skip older versions. If you do really want to publish both an older version and a newer version of the same package, I think you'll need to upload the old version manually, or open a feature request in this repo.

**Comment by vince-fugnitto** [comment]
> Then please also specify the relevant "checkout" location, e.g. like so:

@jankeromnes I updated the code to include the `checkout` (tag of the specific release version).

File: extensions.json (hunk 3 above)
Comment (jankeromnes) [reply extensions.json]: See https://github.com/open-vsx/publish-extensions/pull/16#issuecomment-603300080

File: extensions.json (hunk 2 above)
Comment (jankeromnes) [reply extensions.json]: See https://github.com/open-vsx/publish-extensions/pull/16#issuecomment-603300080

File: extensions.json (hunk 1 above)
Comment (jankeromnes) [reply extensions.json]: See https://github.com/open-vsx/publish-extensions/pull/16#issuecomment-603300080

**Comment by vince-fugnitto** [comment]
@jankeromnes @akosyakov 
I have a question, do releases from these repositories we reference still need to be published as `.vsix`? In the pull-request I referenced the upstream repos (they do not have `.vsix`) and not our own forks which we release ourselves.

**Comment by marcdumais-work** [comment]
> @jankeromnes @akosyakov
> I have a question, do releases from these repositories we reference still need to be published as `.vsix`? 

I think so

> In the pull-request I referenced the upstream repos (they do not have `.vsix`) and not our own forks which we release ourselves.

The `.vsix` are built from sources and then published.

**Comment by jankeromnes** [comment]
@vince-fugnitto: @marcdumais-work is right, upstream repositories don't need to publish. `.vsix` files, because `ovsx` will build these from the Git checkout.

FYI, the [publishing process](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L57-L82) can be summarized like this:

1. [`git clone <repository>`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L58)
2. ([`git checkout <checkout>`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L60) if requested)
3. [`npm install` or `yarn install`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L66) (depending on the presence of a `yarn.lock` file)
4. ([`ovsx create-namespace <publisher>`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L68-L74) if needed)
5. [`ovsx publish`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L76-L82) (with `--yarn` if a `yarn.lock` file was detected earlier)

Also, a quick update from [the latest CI run](https://github.com/open-vsx/publish-extensions/runs/532380660):

- ❌ Publishing `ms-vscode.node-debug` failed because it depends on `ms-vscode.node-debug2` (but publishing should work in the next CI run, see below):

```
[FAIL] Could not process extension: {
  "id": "ms-vscode.node-debug",
  "version": "1.35.3",
  "repository": "https://github.com/microsoft/vscode-node-debug",
  "checkpoint": "v1.35.3"
}
Error: Cannot resolve dependency: ms-vscode.node-debug2
    at Object.<anonymous> (/home/runner/work/publish-extensions/publish-extensions/node_modules/ovsx/lib/publish.js:52:19)
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/publish-extensions/publish-extensions/node_modules/ovsx/lib/publish.js:14:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

- ✅ Publishing `ms-vscode.node-debug2` worked

```
🚀  Published ms-vscode.node-debug2 v1.33.0
[OK] Successfully published ms-vscode.node-debug2 to Open VSX!
```

- ❌ Publishing `ms-vscode.references-view` didn't work:

```
FAIL] Could not process extension: {
  "id": "ms-vscode.references-view",
  "version": "0.0.4",
  "repository": "https://github.com/microsoft/vscode-references-view",
  "checkout": "0.0.4"
}
Error: Command failed: npm install

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/flatmap-stream/-/flatmap-stream-0.1.1.tgz
npm ERR! 404 
npm ERR! 404  'flatmap-stream@0.1.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'repository'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

**Comment by marcdumais-work** [comment]
> 5\. [`ovsx publish`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L76-L82) (with `--yarn` if a `yarn.lock` file was detected earlier)

Hi @jankeromnes - just curious: what changes in the publishing when `--yarn` is used?

**Comment by spoenemann** [comment]
`vsce publish --help` says about that flag:

> Use yarn instead of npm while packing extension files

**Comment by spoenemann** [comment]
I think this is relevant because `vsce` runs the `vscode:prepublish` script before packing.

**Comment by jankeromnes** [comment]
> ❌ Publishing ms-vscode.node-debug failed, because it depends on ms-vscode.node-debug2, but publishing should work in the next CI run

✅ It did indeed in https://github.com/open-vsx/publish-extensions/runs/535352530:

```
🚀  Published ms-vscode.node-debug v1.44.2
[OK] Successfully published ms-vscode.node-debug to Open VSX!
```

**Pull request merged**